src/stage/approved.rs

```diff
@@ -848,16 +786,17 @@ impl ApprovedPeer {
             was_elder = self.chain.is_self_elder();
         }
 
-        Ok(Transition::Stay)
+        Ok(())
     }
 
     fn handle_accumulated_event(
```

Comment:
Interesting, we're having a `Saga-pattern` here. Reacting to an `event` and (in effect) producing a `cmd`. Could be made explicit to clear things up and decouple the code better.